README.md

```diff
@@ -396,14 +397,59 @@ authClient.signIn({
 
 ### `signOut()`
 
-Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example).
+Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example) and clears all tokens stored in the `TokenManager`.
+
+By default:
+
+  - the user will be signed out using an API method 
+  - the page location will not be changed. 
+  
+When calling `signOut` with the default options, it is generally recommended to reload the page after the promise resolves to ensure your app is properly re-initialized in an unauthenticated state.
+
+Since the API method relies upon an XHR call and the ability to set 3rd-party cookies, it may fail in certain scenarios, leaving the user's remote session open. For this reason, we generally recommend using the redirect method over the API method. The redirect method sets the window location to Okta where the session and associated cookies will be fully cleared.
+
+`signOut` takes the following options:
+
+* `redirect` - (default: false) If `true`, enables the logout redirect method with redirect to an Okta-hosted login page. Users who signin from this page may be sent to their Okta user homepage rather than this specific application. To set an application-specific login page, use the `postLogoutRedirectUri` option
+* `postLogoutRedirectUri` - (default: undefined) Enables the logout redirect method, if not set with the `redirect` option, and also sets a custom URI to which the browser will be redirected after logout. **The URI must be whitelisted** as a `Logout Redirect Uri` in your application's general settings. It is common to use your application's origin URI as the `postLogoutRedirectUri` so that users are sent to the application "home page" after signout.
+* `idToken` - (default: auto) Specifies the idToken. By default, `signOut` will look for a token named `idToken` within the `TokenManager`. If you have stored the idToken in a different location, you should retrieve it first and then pass it here. In any case where the `idToken` is not available, the user will be signed out using the API method.
```

Comment:
The config value `tokenManager.storageKey` is for the entire storage object (the key visible in localStorage). 

In this case 'idToken' is the key the token itself is stored under, within the storage object. AuthJS doesn't offer a config for that value because it never stores the tokens directly, it only exposes the TokenManager. It is the responsibility of the callback component to call `parseFromUrl` and then store the tokens in the TokenManager, using any key they want.

AuthJS also has not (until this PR) attempted to read from the TokenManager, as the `isAuthenticated` is currently implemented in the OIDC sdks.

This auto logic will work with the convention adopted by our existing OIDC sdks.

Reply:
Thanks.  Some thoughts,

Write the signature as `(default: null)` as that's more correct for the signature.  Your following sentences explain the auto behaviour

I think we should to more to explain why ID token may be required (to do the redirect-based logout flow)